core/src/main/scala/org/apache/spark/scheduler/ReplayListenerBus.scala

```diff
@@ -118,8 +118,6 @@ private[spark] class ReplayListenerBus extends SparkListenerBus with Logging {
       case e: HaltReplayException =>
         // Just stop replay.
       case _: EOFException if maybeTruncated =>
-      case _: IOException if maybeTruncated =>
```

Comment:
OK, this could be OK, if this was really added only to address what you are fixing here.